# LABORATORIO DIRIGIDO N.° 02
## Analítica Empresarial Integrada — TECSUP 2026-II

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Perfil analítico comparado de empresas del sector azucarero peruano |
| **N.° de laboratorio** | 2 |
| **Tema** | Estadios de evolución analítica y modelo DELTA de madurez organizacional |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-II |
| **Duración** | 190 minutos (7:00 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Fuente de datos** | SMV y BCRP — datos reales y oficiales |
| **Modalidad** |

---

**Equipo N.°:** _____ &nbsp;&nbsp;&nbsp; **Sección:** 6C28 &nbsp;&nbsp;&nbsp; **Fecha:** 26/08/26

**Apellidos y nombres del estudiante:** Rojas Zuñiga Christopher

**Integrantes del equipo:** _(apellidos y nombres de todos los integrantes)_

---


**Elemento de la capacidad terminal:** comprende los estadios de evolución analítica y las dimensiones del modelo DELTA y de su ampliación DELTA Plus; aplica estos marcos a la comparación de indicadores financieros reales y diferencia evidencia, inferencia y opinión.

> **Alcance.** Se analizan estados financieros individuales y anuales del ejercicio 2024. Los importes de la SMV se expresan en miles y pueden estar reportados en soles o dólares según la empresa. Las razones financieras son comparables cuando el numerador y el denominador pertenecen a la misma empresa, moneda y periodo.


## Organización de la sesión — 7:00 p. m. a 10:10 p. m.

**Duración total:** 190 minutos · **Receso:** 15 minutos · **Trabajo efectivo:** 175 minutos.

| Horario | Tiempo | Desarrollo |
|---|---:|---|
| 7:00–7:10 | 10 min | Apertura del caso y conexión con los estadios de evolución analítica. |
| 7:10–7:35 | 25 min | Actividad 1. Revisión de conceptos. Modelos DELTA y DELTA Plus. |
| 7:35–8:00 | 25 min | Actividad 2. Entorno y descarga de información financiera de la SMV. |
| 8:00–8:30 | 30 min | Actividad 2. Normalización, validación e indicadores financieros. Reto 1. |
| 8:30–8:45 | 15 min | **RECESO** |
| 8:45–9:15 | 30 min | Actividad 2. Comparación con Polars y DuckDB. Visualización. Reto 2. |
| 9:15–9:35 | 20 min | Actividad 2. Contexto sectorial del BCRP. Reto 3. |
| 9:35–10:00 | 25 min | **Reto de aplicación y retroalimentación.** Ejercicios 1 a 5. |
| 10:00–10:10 | 10 min | Actividad 3. Informe, conclusiones y preguntas de DELTA Plus. |

> **Regla de trabajo:** no avance de bloque sin registrar la interpretación solicitada. El objetivo no es ejecutar celdas, sino convertir datos en evidencia para una decisión.


## Actividad 1 — Revisión de conceptos (25 minutos)

**Propósito.** Consolidar el vocabulario técnico de la semana antes del desarrollo práctico. La precisión conceptual constituye un requisito del análisis: un indicador correctamente calculado pierde valor si se interpreta con categorías imprecisas.

**Instrucciones.** Complete la tabla con definiciones elaboradas con sus propias palabras. No se admite la reproducción literal de fuentes externas ni de sistemas generativos. La columna de la derecha contiene una pregunta de apoyo: si su definición permite responderla, la definición es suficiente; si no lo permite, corríjala antes de continuar.

**Evidencia esperada.** Tabla completa con las definiciones registradas.

| Concepto | Definición elaborada por el estudiante | Pregunta de apoyo |
|---|---|---|
| Estadios de la evolución analítica | Son los niveles por los que pasa una empresa al aprender a usar datos. Al principio solo mira lo que ya pasó (estadio 2), y después empieza a predecir lo que va a pasar (estadio 3). | ¿Qué distingue a una organización del estadio 2 de una del estadio 3? |
| Modelo DELTA | Es un esquema que dice que para triunfar con datos, la empresa necesita mejorar todo al mismo tiempo: datos, personas, tecnología, estrategia, etc. Si una parte falla, todo se debilita. | ¿Por qué el modelo sostiene que las dimensiones deben avanzar de manera conjunta? |
| Modelo DELTA Plus | Es una versión mejorada del DELTA original, que añade cosas importantes como la ética y el control de riesgos, que el primero no tomaba en cuenta. | ¿Qué limitación del modelo original corrige la ampliación? |
| Dimensión Technology | Es la parte de herramientas y sistemas. Una empresa puede tener buenos datos y buenos analistas, pero si sus programas son viejos o no se conectan entre sí, su madurez sigue siendo baja. | ¿Puede una organización con datos y analistas presentar madurez limitada? Justifique. |
| Dimensión Analytical Techniques | Son los métodos para analizar datos. La técnica descriptiva solo dice "qué pasó" (ej. ventas de ayer). La diagnóstica va más allá y dice "por qué pasó" (ej. las ventas bajaron porque llovió). | ¿Qué diferencia una técnica descriptiva de una diagnóstica? Proponga un ejemplo. |
| Organización que compite con analítica | Es una empresa que usa los datos para ganar mercado, no solo para hacer gráficos bonitos. No basta con tener paneles; hay que usarlos para tomar decisiones y actuar. | ¿Basta con disponer de tableros de control para competir con analítica? |
| Ventaja competitiva basada en datos | Es la capacidad de hacer mejores cosas que los rivales porque manejas mejor la información. Es difícil de copiar porque no es solo tecnología, es también la forma de pensar y trabajar de la gente. | ¿Por qué razones esta ventaja puede resultar difícil de imitar por un competidor? |
| Impacto de la IA generativa en la empresa | La IA ayuda a escribir, resumir y crear contenido, pero hay tareas delicadas, como decidir si un préstamo es seguro o interpretar una ley, que no se le deben dejar a la máquina. | ¿Qué tarea del análisis financiero no debería delegarse a un sistema generativo? |
| Perfil analítico comparado | Es el conjunto de características que se miran para poder comparar dos empresas. Para que la comparación sea válida, ambas tienen que ser parecidas: mismo tamaño, mismo mercado, mismas reglas contables. | ¿Qué condiciones deben cumplirse para que dos empresas sean comparables? |
| Indicador financiero como evidencia | Un número puede estar bien calculado, pero si no se usa en su contexto correcto, no sirve como prueba. Por ejemplo, comparar ventas de enero sin saber que hubo un feriado que afectó. | ¿En qué caso un indicador correctamente calculado no constituye evidencia válida? |

**Criterio de cierre.** No se avanza al desarrollo práctico mientras existan conceptos sin definición registrada.


## Marco conceptual — Del modelo DELTA al modelo DELTA Plus

El modelo DELTA fue formulado por Davenport, Harris y Morison en *Analytics at Work* (2010) como un instrumento para evaluar la madurez analítica de una organización a partir de cinco dimensiones: **Data**, referida a la disponibilidad, integración y calidad de los datos; **Enterprise**, que examina si la capacidad analítica se encuentra dispersa en áreas aisladas o coordinada a escala institucional; **Leadership**, que valora el compromiso efectivo de la dirección con la decisión basada en evidencia; **Targets**, que analiza si los esfuerzos analíticos se concentran en los objetivos de mayor valor para el negocio; y **Analysts**, que considera la disponibilidad de profesionales con las competencias requeridas. El principio central del modelo establece que estas dimensiones deben progresar de manera conjunta, dado que un avance aislado en una de ellas no eleva la madurez global de la organización.

Con posterioridad, y ante la transformación del entorno tecnológico de la analítica empresarial, el modelo fue ampliado bajo la denominación **DELTA Plus**, que incorpora dos dimensiones adicionales. La primera es **Technology**, que comprende la infraestructura y las plataformas que permiten desarrollar y desplegar capacidades analíticas: soluciones en la nube, *data warehouses*, *data lakes* y arquitecturas *lakehouse*, entornos de procesamiento de grandes volúmenes de información, herramientas de visualización, plataformas analíticas corporativas, componentes de código abierto e infraestructura para el desarrollo de modelos de aprendizaje automático e inteligencia artificial. Su incorporación obedece a una constatación de orden práctico: una organización puede disponer de datos pertinentes y de analistas calificados y presentar, aun así, una madurez analítica limitada si carece de la tecnología necesaria para almacenar, procesar, analizar y poner en uso esa información de manera periódica y reproducible.

La segunda dimensión añadida es **Analytical Techniques**, que evalúa la variedad y la sofisticación de las técnicas empleadas para convertir los datos en información útil para la decisión. Estas técnicas admiten una progresión reconocible: la analítica **descriptiva** establece qué ocurrió; la **diagnóstica** indaga por qué ocurrió; la **predictiva** estima qué podría ocurrir; y la **prescriptiva** determina qué conviene hacer. Sobre esa base se incorporan progresivamente el aprendizaje automático, las técnicas de segmentación, los modelos de clasificación, los pronósticos, los métodos de optimización y las aplicaciones de inteligencia artificial. Dichas técnicas se presentan aquí como ilustración de la evolución de la capacidad analítica de una organización y serán objeto de desarrollo en semanas posteriores del curso; el presente laboratorio se sitúa en los niveles descriptivo y diagnóstico.

### Tabla comparativa

| | Modelo DELTA (2010) | Modelo DELTA Plus |
|---|---|---|
| **Dimensiones** | Cinco | Siete |
| **Componentes** | Data, Enterprise, Leadership, Targets, Analysts | Las cinco anteriores, más Technology y Analytical Techniques |
| **Objeto de evaluación** | Condiciones organizacionales de la analítica | Condiciones organizacionales y capacidad técnica de ejecución |
| **Pregunta que responde** | ¿Dispone la organización de datos, dirección, objetivos y personas? | Además de lo anterior, ¿cuenta con la tecnología y las técnicas para ejecutarlo de manera sostenida? |
| **Limitación que corrige** | — | Una organización con datos y analistas puede presentar madurez limitada por ausencia de tecnología o por el uso de técnicas elementales |

> **Precisión conceptual.** DELTA y DELTA Plus no constituyen modelos alternativos. DELTA Plus es una ampliación del modelo original: conserva íntegramente sus cinco dimensiones y añade dos. Al emitir un diagnóstico debe indicarse explícitamente cuál de los dos marcos se está aplicando.


## Actividad 2 — Desarrollo práctico y ejecución

**Propósito:** descargar información financiera anual y real de empresas del sector azucarero peruano, construir un perfil comparado de rentabilidad, liquidez y endeudamiento, y relacionarlo prudentemente con el contexto sectorial del BCRP mediante pandas, Polars, DuckDB y Plotly.

**Empresas:** Casa Grande S.A.A., Cartavio S.A.A., Agroindustrial Laredo S.A.A. y Empresa Agroindustrial Pomalca S.A.A.


## Paso 1: Preparación del entorno


In [1]:
# Ejecute esta celda en Google Colab. Las versiones se fijan para favorecer la reproducibilidad.
%pip install -q polars==1.17.1 duckdb==1.1.3 econdata==1.0.17

import json, html, re, unicodedata
from xml.etree import ElementTree as ET

import requests
import pandas as pd
import numpy as np
import polars as pl
import duckdb
import plotly.graph_objects as go
from IPython.display import display
from econdata import BCRP

pd.set_option("display.width", 160)
pd.set_option("display.max_columns", 40)
print("Entorno listo.")


   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.4/36.4 MB 26.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.2/20.2 MB 45.0 MB/s eta 0:00:00
Entorno listo.


## Paso 2: Descarga de información financiera real desde la SMV

Se emplea el servicio web oficial de la SMV para consultar las principales cuentas y el Estado de Situación Financiera. Las operaciones utilizadas son `obtener_InfoFinanciera` y `obtener_BalanceGeneral`. El nombre oficial `obtener_GanciaPerdida` también se conserva para consultas del Estado de Resultados.


In [2]:
SERVICIO_SMV = "https://mvnet.smv.gob.pe/ws_od_eeff/WebServiceInfoFinanciera.asmx"
EJERCICIO = 2024
PERIODO = "A"   # Anual
TIPO = "I"      # Individual
AZUCARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

def etiqueta_local(etiqueta):
    '''Retira el espacio de nombres de una etiqueta XML.'''
    return str(etiqueta).split("}")[-1]

def payload_a_dataframe(payload):
    '''Convierte el contenido JSON o XML incluido en la respuesta SOAP en DataFrame.'''
    payload = html.unescape(str(payload)).strip()
    try:
        objeto = json.loads(payload)
        if isinstance(objeto, dict):
            objeto = objeto.get("rows", objeto.get("data", objeto))
        if isinstance(objeto, dict):
            objeto = [objeto]
        return pd.DataFrame(objeto)
    except json.JSONDecodeError:
        raiz = ET.fromstring(payload)
        registros = []
        for nodo in raiz.iter():
            hijos = list(nodo)
            if len(hijos) >= 5 and all(not list(hijo) for hijo in hijos):
                registros.append({etiqueta_local(h.tag): h.text for h in hijos})
        return pd.DataFrame(registros)

def descargar_smv(operacion, ejercicio=EJERCICIO, periodo=PERIODO, tipo=TIPO):
    '''Consulta una operación oficial del servicio SOAP de la SMV.'''
    sobre = f'''<?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
      <soap:Body><{operacion} xmlns="http://tempuri.org/">
        <Ejercicio>{ejercicio}</Ejercicio><Periodo>{periodo}</Periodo><Tipo>{tipo}</Tipo>
      </{operacion}></soap:Body>
    </soap:Envelope>'''
    respuesta = requests.post(
        SERVICIO_SMV,
        data=sobre.encode("utf-8"),
        timeout=90,
        headers={
            "Content-Type": "text/xml; charset=utf-8",
            "SOAPAction": f'"http://tempuri.org/{operacion}"',
        },
    )
    respuesta.raise_for_status()
    raiz = ET.fromstring(respuesta.content)
    resultado = next(
        (n for n in raiz.iter() if etiqueta_local(n.tag) == f"{operacion}Result"),
        None,
    )
    if resultado is None or not (resultado.text or "").strip():
        raise ValueError(f"La SMV no devolvió datos para {operacion}.")
    datos = payload_a_dataframe(resultado.text)
    if datos.empty:
        raise ValueError(f"La respuesta de {operacion} no contiene registros.")
    return datos

principales = descargar_smv("obtener_InfoFinanciera")
balance = descargar_smv("obtener_BalanceGeneral")

print("Principales cuentas:", principales.shape)
print("Estado de situación financiera:", balance.shape)
display(principales.head(3))


Principales cuentas: (276, 16)
Estado de situación financiera: (20574, 15)


,RPJ,TipoEmpresa,TipoSector,NombreEmpresa,RUC,CIIU,Ejercicio,TipoInformacion,Trimestre,Moneda,MetodoFlujoEfectivo,ActivoTotal,PatrimonioTotal,TotalIngreso,UtilidadNeta,PasivoTotal
0,L00474,EMPRESAS MERCADO ALTERNATIVO DE VALORES,DIVERSOS,A. JAIME ROJAS REPRESENTACIONES GENERALES S.A.,20102032951,5190,2024,Anual Individual,Anual,Soles,M todo Directo,77710,39497,77196,6786,38213
1,I00004,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,AC CAPITALES SOCIEDAD ADMINISTRADORA DE FONDOS...,20504893295,6430,2024,Anual Individual,Anual,Soles,M todo Directo,9070,7373,4880,-85,1697
2,OE7511,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,ACRES SOCIEDAD ADMINISTRADORA DE FONDOS DE INV...,20601498996,6430,2024,Anual Individual,Anual,Soles,M todo Indirecto,2724,2238,2348,176,486


## Paso 3: Normalización, filtrado y validación del sector

Se normalizan los nombres de las columnas, se filtran las cuatro empresas y se verifica que todas estén representadas. No se utiliza una base simulada ni un respaldo inventado.


In [3]:
def nombre_columna(texto):
    texto = re.sub(r"(?<!^)(?=[A-Z])", "_", str(texto))
    texto = unicodedata.normalize("NFKD", texto).encode("ascii", "ignore").decode()
    return re.sub(r"[^a-z0-9]+", "_", texto.lower()).strip("_")

def preparar(df):
    df = df.copy()
    df.columns = [nombre_columna(c) for c in df.columns]
    return df

def resolver_columna(df, candidatos):
    '''Encuentra una columna aun cuando la SMV cambie mayúsculas o convenciones de nombre.'''
    disponibles = {nombre_columna(c): c for c in df.columns}
    for candidato in candidatos:
        clave = nombre_columna(candidato)
        if clave in disponibles:
            return disponibles[clave]
    raise KeyError(f"No se encontró ninguna de estas columnas: {candidatos}. Disponibles: {list(df.columns)}")

def estandarizar_empresa(df):
    df = preparar(df)
    origen = resolver_columna(df, ["nombre_empresa", "razon_social", "empresa"])
    return df.rename(columns={origen: "nombre_empresa"})

def filtrar_azucareras(df):
    patron = "|".join(map(re.escape, AZUCARERAS))
    return df[
        df["nombre_empresa"].astype(str).str.upper().str.contains(patron, regex=True, na=False)
    ].copy()

principales = filtrar_azucareras(estandarizar_empresa(principales))
balance = filtrar_azucareras(estandarizar_empresa(balance))

empresas = sorted(principales["nombre_empresa"].dropna().unique())
print("Empresas encontradas:")
for empresa in empresas:
    print("-", empresa)

faltantes = [nombre for nombre in AZUCARERAS if not any(nombre in e.upper() for e in empresas)]
if faltantes:
    raise ValueError(f"No se localizaron todas las empresas. Faltantes: {faltantes}")


Empresas encontradas:
- AGROINDUSTRIAL LAREDO S.A.A.
- CARTAVIO SOCIEDAD ANONIMA ABIERTA
- CASA GRANDE SOCIEDAD ANONIMA ABIERTA
- EMPRESA AGROINDUSTRIAL POMALCA S.A.A.


### Paso 3: La cuenta correcta no se busca por su nombre

El Estado de Situación Financiera trae 69 cuentas por empresa, y **más de una contiene el texto «Activos Corrientes»**, pero solo una es el total. Filtrar por la descripción es un error frecuente y silencioso: no lanza ninguna excepción, devuelve un número, y ese número está mal.

Ejecute la celda siguiente y observe el catálogo: la cuenta `1D0122` —«Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral»— vale 0 y, de usarse, produce una razón corriente igual a cero.


In [4]:
# Catálogo de cuentas cuyo texto contiene "corriente", para una de las empresas.
muestra = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])
catalogo = muestra[
    muestra[columna_desc].astype(str).str.contains("orriente", na=False)
][["cuenta", columna_desc, "monto1"]]
print("Empresa de referencia:", empresas[0])
display(catalogo.reset_index(drop=True))

# Los dos totales corrientes se declaran por su CÓDIGO oficial, no por su
# descripción. Observe en el catálogo de arriba por qué filtrar por texto falla.
TOTAL_ACTIVO_CORRIENTE = "1D01ST"
TOTAL_PASIVO_CORRIENTE = "1D03ST"

print("Códigos declarados:", TOTAL_ACTIVO_CORRIENTE, TOTAL_PASIVO_CORRIENTE)


Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0122,Activos Corrientes Distintos al Efectivo Pigno...,0.0
1,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
2,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
3,1D01ST,Total Activos Corrientes,137966.0
4,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
5,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
6,1D02ST,Total Activos No Corrientes,802358.0
7,1D0315,Total de Pasivos Corrientes Distintos de Pasiv...,125303.0
8,1D03ST,Total Pasivos Corrientes,125303.0
9,1D0412,"Pasivos por Impuestos Corrientes, no Corriente",0.0


Códigos declarados: 1D01ST 1D03ST


## Paso 4: Construcción reproducible de los indicadores financieros

| Familia | Indicador | Fórmula |
|---|---|---|
| Rentabilidad | Margen neto | Utilidad neta / Ingresos |
| Rentabilidad | ROA | Utilidad neta / Activo total |
| Rentabilidad | ROE | Utilidad neta / Patrimonio |
| Liquidez | Razón corriente | Activo corriente / Pasivo corriente |
| Endeudamiento | Endeudamiento | Pasivo total / Activo total |
| Endeudamiento | Apalancamiento | Pasivo total / Patrimonio |

La razón corriente se obtiene del Estado de Situación Financiera mediante las descripciones oficiales de activos y pasivos corrientes.


In [5]:
def renombrar_primera(df, destino, candidatos):
    if destino in df.columns:
        return df
    return df.rename(columns={resolver_columna(df, candidatos): destino})

# Estandarización de las cuentas principales que devuelve obtener_InfoFinanciera.
for destino, candidatos in {
    "ventas": ["total_ingreso", "total_ingresos", "ingresos", "ventas_netas"],
    "utilidad_neta": ["utilidad_neta", "ganancia_perdida_neta", "resultado_neto"],
    "activo_total": ["activo_total", "total_activo", "total_activos"],
    "pasivo_total": ["pasivo_total", "total_pasivo", "total_pasivos"],
    "patrimonio": ["patrimonio_total", "total_patrimonio", "patrimonio"],
}.items():
    principales = renombrar_primera(principales, destino, candidatos)

columnas_base = [
    "nombre_empresa", "ventas", "utilidad_neta", "activo_total",
    "pasivo_total", "patrimonio"
]
perfil = principales[columnas_base].drop_duplicates("nombre_empresa").copy()
perfil = perfil.rename(columns={"nombre_empresa": "empresa"})

def a_numero(valor):
    '''Convierte importes de la SMV aunque incluyan separadores, espacios o paréntesis.'''
    if pd.isna(valor):
        return np.nan
    if isinstance(valor, (int, float, np.number)):
        return float(valor)
    texto = str(valor).strip().replace("\u00a0", "").replace(" ", "")
    negativo = texto.startswith("(") and texto.endswith(")")
    texto = texto.strip("()")
    # La SMV normalmente usa punto decimal. Las comas se eliminan como separador de miles.
    texto = texto.replace(",", "")
    texto = re.sub(r"[^0-9.\-]", "", texto)
    if texto in {"", "-", "."}:
        return np.nan
    numero = float(texto)
    return -numero if negativo else numero

for columna in ["ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio"]:
    perfil[columna] = perfil[columna].map(a_numero)

# Las cuentas se identifican por su CÓDIGO oficial y no por su descripción.
# El código es estable entre ejercicios y evita capturar cuentas cuyo texto
# también contiene "Activos Corrientes" sin ser el total: por ejemplo 1D0122,
# "Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral",
# que vale 0 y, de usarse, produce una razón corriente igual a cero.
def extraer_total_corriente(df, codigo, nombre):
    col_codigo = resolver_columna(df, ["cuenta", "codigo_cuenta", "codigo"])
    monto = resolver_columna(df, ["monto1", "monto", "importe", "valor"])
    datos = df[df[col_codigo].astype(str).str.strip() == codigo][
        ["nombre_empresa", monto]].copy()
    datos[monto] = datos[monto].map(a_numero)
    datos = datos.dropna().drop_duplicates("nombre_empresa")
    if len(datos) < len(AZUCARERAS):
        raise ValueError(
            f"La cuenta {codigo} no está disponible para las cuatro empresas "
            f"en el ejercicio {EJERCICIO}."
        )
    return datos.rename(columns={"nombre_empresa": "empresa", monto: nombre})

perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_CORRIENTE, "activo_corriente"),
    on="empresa", how="inner")
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_PASIVO_CORRIENTE, "pasivo_corriente"),
    on="empresa", how="inner")

if len(perfil) < 4:
    raise ValueError("No fue posible construir el perfil completo de las cuatro empresas.")

columnas_calculo = [
    "ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio",
    "activo_corriente", "pasivo_corriente"
]
faltantes = perfil[columnas_calculo].isna().sum()
if faltantes.any():
    raise ValueError(
        "Existen cuentas sin valor numérico: " +
        ", ".join(f"{c} ({n})" for c, n in faltantes.items() if n)
    )

if (perfil[["ventas", "activo_total", "patrimonio", "pasivo_corriente"]] == 0).any().any():
    raise ValueError("Existe un denominador igual a cero; revise las cuentas antes de calcular.")

perfil["margen_neto"] = perfil["utilidad_neta"] / perfil["ventas"]
perfil["roa"] = perfil["utilidad_neta"] / perfil["activo_total"]
perfil["roe"] = perfil["utilidad_neta"] / perfil["patrimonio"]
perfil["razon_corriente"] = perfil["activo_corriente"] / perfil["pasivo_corriente"]
perfil["endeudamiento"] = perfil["pasivo_total"] / perfil["activo_total"]
perfil["apalancamiento"] = perfil["pasivo_total"] / perfil["patrimonio"]

numeros = perfil.select_dtypes(include="number").to_numpy()
if np.isinf(numeros[~np.isnan(numeros)]).any():
    raise ValueError("Se obtuvo una división entre cero; revise las cuentas de la empresa afectada.")

display(perfil.round(3))


,empresa,ventas,utilidad_neta,activo_total,pasivo_total,patrimonio,activo_corriente,pasivo_corriente,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,55453.0,940324.0,327342.0,612982.0,137966.0,125303.0,0.126,0.059,0.090,1.101,0.348,0.534
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,91846.0,926161.0,542687.0,383474.0,294933.0,307537.0,0.172,0.099,0.240,0.959,0.586,1.415
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,73490.0,1923732.0,874083.0,1049649.0,429304.0,362479.0,0.115,0.038,0.070,1.184,0.454,0.833
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,9636.0,1352469.0,679594.0,672875.0,168835.0,108618.0,0.041,0.007,0.014,1.554,0.502,1.010


**Pregunta 1.** Identifique la empresa con mayor ROE y la empresa con mayor endeudamiento. ¿Coinciden? Interprete la relación observada sin afirmar causalidad..

**Respuesta:** Cartavio es la empresa que alcanza el ROE más elevado (24.0%) y, al mismo tiempo, tiene el mayor nivel de endeudamiento (58.6%). Por eso, los dos resultados corresponden a la misma empresa. Desde mi interpretación, esto puede estar relacionado con el uso de deuda dentro de su estructura financiera para aumentar el rendimiento de los accionistas. Sin embargo, con estos datos por sí solos no se puede concluir que la deuda sea la razón directa de su mayor rentabilidad.



### Reto 1 — Predicción antes del ranking (5 minutos)

**Sin ordenar todavía la tabla**, responda en equipo:

1. ¿Qué empresa creen que presenta el mejor perfil financiero?

**Respuesta** : Cartavio

2. Sustenten la predicción con **dos indicadores** como mínimo.

**Respuesta:** Con datos previos ya sabíamos que Cartavio tenía la mayor rotación de activos y el mayor margen neto del grupo, dos indicadores que anticipan una rentabilidad superior.

3. ¿Qué indicador podría hacer cambiar su conclusión?

**Respuesta:** El endeudamiento — si se prioriza solidez financiera sobre rentabilidad pura, Cartavio deja de ser la "mejor" opción porque tiene el mayor apalancamiento del grupo.

**Predicción del equipo:**

**Justificación:**

Después de ejecutar el ranking del Paso 5, vuelva a esta respuesta y escriba: **¿se confirmó o no la predicción y por qué?**

**Respuesta:** Sí, se confirmó en ROE (Cartavio lidera con 24.0%), pero con un matiz: Cartavio no lidera en liquidez (razón corriente 0.96, la más baja del grupo) ni en endeudamiento (el más alto), por lo que "mejor perfil financiero" depende de qué dimensión se priorice.

## Paso 5: Comparación sectorial con Polars y DuckDB


In [6]:
perfil_pl = pl.from_pandas(perfil)

ranking_roe = (
    perfil_pl
    .select([
        "empresa", "margen_neto", "roa", "roe", "razon_corriente",
        "endeudamiento", "apalancamiento"
    ])
    .sort("roe", descending=True)
)
ranking_roe


empresa,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
str,f64,f64,f64,f64,f64,f64
"""CARTAVIO SOCIEDAD ANONIMA ABIE…",0.171651,0.099169,0.23951,0.959016,0.585953,1.415186
"""AGROINDUSTRIAL LAREDO S.A.A.""",0.126336,0.058972,0.090464,1.101059,0.348116,0.534016
"""CASA GRANDE SOCIEDAD ANONIMA A…",0.115246,0.038202,0.070014,1.184356,0.454368,0.832738
"""EMPRESA AGROINDUSTRIAL POMALCA…",0.041172,0.007125,0.014321,1.554392,0.502484,1.009986


In [7]:
# Misma comparación del ranking anterior, ahora expresada en SQL sobre DuckDB.
# Ejecutar la misma pregunta en dos motores distintos y obtener el mismo
# resultado es una comprobación de reproducibilidad, no una coincidencia.
CONSULTA = """
SELECT empresa,
       ROUND(margen_neto * 100, 1)    AS margen_neto_pct,
       ROUND(roa * 100, 1)            AS roa_pct,
       ROUND(roe * 100, 1)            AS roe_pct,
       ROUND(razon_corriente, 2)      AS razon_corriente,
       ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct,
       ROUND(apalancamiento, 2)       AS apalancamiento
FROM perfil
ORDER BY roe_pct DESC
"""

consulta = duckdb.sql(CONSULTA).df()
display(consulta)

# Comprobación automática: DuckDB y Polars deben coincidir.
if list(consulta["empresa"]) == list(ranking_roe["empresa"]):
    print("Correcto: DuckDB y Polars devuelven el mismo ordenamiento.")
else:
    print("Los ordenamientos no coinciden. Revise las columnas y el ORDER BY.")


,empresa,margen_neto_pct,roa_pct,roe_pct,razon_corriente,endeudamiento_pct,apalancamiento
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,17.2,9.9,24.0,0.96,58.6,1.42
1,AGROINDUSTRIAL LAREDO S.A.A.,12.6,5.9,9.0,1.10,34.8,0.53
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,11.5,3.8,7.0,1.18,45.4,0.83
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4.1,0.7,1.4,1.55,50.2,1.01


Correcto: DuckDB y Polars devuelven el mismo ordenamiento.


In [8]:
# Rankings complementarios para comparar dimensiones distintas del perfil.
ranking_comparado = pd.DataFrame({
    "empresa": perfil["empresa"],
    "rank_roe": perfil["roe"].rank(ascending=False, method="min").astype(int),
    "rank_liquidez": perfil["razon_corriente"].rank(ascending=False, method="min").astype(int),
    # En endeudamiento, un valor menor recibe mejor posición solo como criterio descriptivo de exposición financiera.
    "rank_endeudamiento": perfil["endeudamiento"].rank(ascending=True, method="min").astype(int),
})
ranking_comparado["cambio_max_posicion"] = (
    ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].max(axis=1)
    - ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].min(axis=1)
)
display(ranking_comparado.sort_values(["rank_roe", "empresa"]))


,empresa,rank_roe,rank_liquidez,rank_endeudamiento,cambio_max_posicion
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,1,4,4,3
0,AGROINDUSTRIAL LAREDO S.A.A.,2,3,1,2
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,3,2,2,1
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4,1,3,3


### Reto 2 — Evidencia, inferencia y conclusión (5 minutos)

Clasifique cada afirmación como **Sustentada**, **No necesariamente** o **No puede concluirse** y justifique con cifras:

| Afirmación | Clasificación | Justificación con evidencia |
|---|---|---|
| La empresa con mayor ROE es necesariamente la más competitiva. | No necesariamente | Cartavio tiene el mayor ROE (24.0%) pero también el mayor endeudamiento (58.6%) y la menor liquidez (razón corriente 0.96 < 1), lo que indica mayor riesgo financiero, no necesariamente mejor posición competitiva integral. |
| Una empresa con mayor endeudamiento posee necesariamente menor madurez analítica. | No puede concluirse | El endeudamiento es un indicador financiero (estructura de capital); los estados financieros no contienen información sobre liderazgo, analistas o tecnología de la empresa, dimensiones necesarias para evaluar madurez analítica según DELTA. |
| Obtener un ordenamiento consistente con Polars y DuckDB fortalece la reproducibilidad del análisis. | Sustentada | Ambos motores, aplicados sobre los mismos datos con la misma lógica de ordenamiento, produjeron el mismo resultado (verificado en el Paso 5), lo cual es evidencia directa de reproducibilidad técnica — aunque no valida ni implica nada sobre madurez organizacional. |

**Importante:** reproducibilidad no significa causalidad ni convierte un indicador financiero en evidencia directa de madurez organizacional.


**Pregunta 2.** Compare los rankings por rentabilidad, liquidez y endeudamiento. Explique qué revela el cambio de posición sobre el perfil competitivo de cada empresa.

**Respuesta:** Cartavio ocupa el primer puesto en rentabilidad, pero queda en el cuarto lugar tanto en liquidez como en endeudamiento, mostrando uno de los cambios más marcados del grupo. Para mí, esto refleja una empresa rentable, aunque con una posición financiera que implica mayor exposición al riesgo y menor margen de liquidez. Laredo destaca por combinar una buena rentabilidad (puesto 2) con el menor endeudamiento (puesto 1), por lo que presenta el balance más estable. Pomalca, en cambio, mantiene resultados débiles en la mayoría de indicadores, aunque sobresale relativamente en liquidez.



## Paso 6: Visualización del perfil comparado


In [9]:
indicadores_pct = ["margen_neto", "roa", "roe", "endeudamiento"]
nombres_indicadores = {
    "margen_neto": "Margen neto",
    "roa": "ROA",
    "roe": "ROE",
    "endeudamiento": "Endeudamiento",
}
perfil_grafico = perfil.copy()
perfil_grafico["empresa_corta"] = (
    perfil_grafico["empresa"].str.upper()
    .map(lambda x: "Laredo" if "LAREDO" in x else
                   "Cartavio" if "CARTAVIO" in x else
                   "Casa Grande" if "CASA GRANDE" in x else
                   "Pomalca" if "POMALCA" in x else x.title())
)
fig = go.Figure()
for indicador in indicadores_pct:
    fig.add_trace(go.Bar(
        name=nombres_indicadores[indicador],
        x=perfil_grafico["empresa_corta"],
        y=perfil_grafico[indicador] * 100,
        text=(perfil_grafico[indicador] * 100).round(1),
        texttemplate="%{text:.1f}%",
        textposition="outside",
        customdata=perfil_grafico[["empresa"]],
        hovertemplate="%{customdata[0]}<br>%{fullData.name}: %{y:.2f}%<extra></extra>",
    ))
fig.update_layout(
    title=f"Perfil financiero comparado — sector azucarero ({EJERCICIO})",
    barmode="group",
    yaxis_title="Porcentaje (%)",
    height=540,
    width=1050,
    template="plotly_white",
    margin=dict(t=110, b=90, l=80, r=40),
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="left", x=0),
)
fig.show()

fig_liquidez = go.Figure(go.Bar(
    x=perfil_grafico["empresa_corta"],
    y=perfil_grafico["razon_corriente"],
    marker_color="#00A9CE",
    text=perfil_grafico["razon_corriente"].round(2),
    texttemplate="%{text:.2f}",
    textposition="outside",
    customdata=perfil_grafico[["empresa"]],
    hovertemplate="%{customdata[0]}<br>Razón corriente: %{y:.2f}<extra></extra>",
))
fig_liquidez.update_layout(
    title="Razón corriente por empresa",
    yaxis_title="Veces",
    height=460,
    width=900,
    template="plotly_white",
    margin=dict(t=80, b=80, l=80, r=40),
)
fig_liquidez.update_yaxes(rangemode="tozero")
fig_liquidez.show()


## Paso 7: Contexto sectorial con datos específicos del BCRP

Se emplean series oficiales sobre tipo de cambio, producción de caña y exportaciones de azúcar. Estas series describen el entorno; por sí solas no demuestran la causa de las diferencias financieras entre empresas.


In [10]:
SERIES = {
    "PN01209PM": "tipo_cambio",
    "PN01790AM": "produccion_cana_azucar_miles_t",
    "PN38774BM": "export_azucar_fob_millones_usd",
    "PN38775BM": "volumen_exportado_azucar",
}

contexto = BCRP.get_data(SERIES, "2020-01", "2024-12")
print("Datos del BCRP:", contexto.shape)
display(contexto.tail(12))


Datos del BCRP: (60, 4)


,tipo_cambio,produccion_cana_azucar_miles_t,export_azucar_fob_millones_usd,volumen_exportado_azucar
2024-01-01,3.743667,755.602441,7.547009,10.321329
2024-02-01,3.830905,814.525062,4.398561,5.993531
2024-03-01,3.712947,563.626363,1.911468,2.355260
2024-04-01,3.718409,634.675782,3.839971,4.951844
2024-05-01,3.735273,812.740966,3.235732,3.843619
2024-06-01,3.789526,830.637431,0.481814,0.618096
2024-07-01,3.769500,840.996455,0.919740,1.246593
2024-08-01,3.745550,1017.287110,1.010892,1.358606
2024-09-01,3.772143,930.472111,3.097093,4.088378
2024-10-01,3.757500,1053.902968,1.906427,2.543543


In [11]:
# Evolución temporal del contexto sectorial. Cada serie se normaliza a índice 100
# para comparar tendencias con unidades diferentes, sin alterar los datos originales.
contexto_grafico = contexto.copy()
columnas_numericas = contexto_grafico.select_dtypes(include="number").columns
base = contexto_grafico[columnas_numericas].apply(lambda s: s.dropna().iloc[0])
indice_100 = contexto_grafico[columnas_numericas].divide(base).multiply(100)

fig_contexto = go.Figure()
for columna in indice_100.columns:
    fig_contexto.add_trace(go.Scatter(
        x=indice_100.index,
        y=indice_100[columna],
        mode="lines",
        name=columna,
    ))
fig_contexto.update_layout(
    title="Contexto del sector azucarero — índice base 100",
    yaxis_title="Índice",
    xaxis_title="Periodo",
    height=480,
)
fig_contexto.show()


**Pregunta 3.** Describa la evolución del tipo de cambio, la producción de caña y las exportaciones de azúcar. Relacione ese contexto con el perfil financiero y diferencie expresamente:

- **Evidencia:** Durante 2024 el tipo de cambio tuvo un comportamiento relativamente estable, moviéndose aproximadamente entre S/ 3.71 y S/ 3.83 por dólar. La producción de caña presentó una evolución creciente, aunque con variaciones, pasando de unas 564 mil toneladas en marzo y superando las 1,000 mil toneladas en algunos meses como agosto y octubre, para terminar el año alrededor de 940 mil toneladas. En las exportaciones de azúcar se observó una fuerte reducción: comenzaron cerca de US$ 7.5 millones en enero y descendieron hasta valores cercanos a US$ 0.5-1 millón entre junio y agosto, para luego recuperarse parcialmente hasta unos US$ 2.1-3.5 millones en noviembre y diciembre.

- **Inferencia:** Considero que el incremento de la producción de caña durante la segunda mitad del año no tuvo un reflejo similar en las exportaciones. Esto podría significar que una mayor parte de la producción fue dirigida al mercado nacional u otros destinos. Además, debido a la poca variación del tipo de cambio, parece poco probable que sus movimientos hayan sido el factor principal detrás de las diferencias financieras observadas entre las empresas.

- **Opinión:** Desde mi punto de vista, la disminución de las exportaciones en varios meses pudo afectar especialmente a las empresas con una mayor dependencia del mercado internacional. Sin embargo, para confirmar esa relación sería necesario conocer cuánto de las ventas de cada empresa corresponde al mercado interno y cuánto al externo. Por ello, relacionar directamente estos datos sectoriales con los resultados financieros de Casa Grande, Cartavio, Laredo y Pomalca debe considerarse una hipótesis y no una conclusión definitiva.


### Reto 3 — Diagnóstico provisional de madurez y DELTA (8 minutos)

Seleccione **una** de las cuatro empresas. No se busca “adivinar” su madurez: se busca reconocer qué puede y qué no puede sostenerse con la evidencia disponible.

| Aspecto | Respuesta del equipo |
|---|---|
| Empresa seleccionada | Cartavio Sociedad Anónima Abierta |
| Evidencia financiera observada | ROE de 24.0% (el más alto del grupo), margen neto de 17.2%, pero razón corriente de 0.96 (menor a 1) y endeudamiento de 58.6% (el más alto del grupo). |
| Estadio de evolución analítica compatible **de manera provisional** | Estadio inicial-intermedio (uso analítico probablemente localizado, no necesariamente extendido a toda la organización) — clasificación tentativa basada únicamente en resultados financieros. |
| Dimensión DELTA evaluable parcialmente | Targets: los buenos resultados de rentabilidad podrían sugerir que la empresa concentra esfuerzos en objetivos de alto valor (margen y eficiencia), aunque esto es una inferencia, no una verificación directa. |
| Dimensiones DELTA que no pueden verificarse | Enterprise, Leadership, Analysts y Technology — ninguna se puede confirmar con estados financieros, ya que requieren observar procesos internos, personal y infraestructura de la empresa. |
| Evidencia organizacional adicional requerida | Entrevistas o documentación sobre uso de herramientas analíticas, estructura del equipo de datos, y evidencia de decisiones estratégicas basadas en analítica (no solo en resultados contables). |

**Cierre del reto:** redacte una frase que distinga claramente **evidencia**, **inferencia** y **opinión**.


### Pausa metodológica — ¿qué NO podemos afirmar?

Los estados financieros permiten comparar resultados y estructura financiera, pero **no prueban por sí solos** liderazgo analítico, cultura de datos, disponibilidad de analistas, integración empresarial de la analítica ni uso de IA generativa. Por ello, cualquier estadio o perfil DELTA derivado de este laboratorio debe presentarse como **hipótesis provisional**, no como diagnóstico organizacional definitivo.


## Actividad 3 — Informe de perfil analítico comparado

Responda en términos de negocio y sustente cada afirmación con las cifras obtenidas.

**A.** Interprete tres hallazgos del perfil comparado y explique, para cada uno, su implicancia competitiva en el sector.

**Respuesta:**

- Cartavio combina el mayor margen neto (17.2%) y mayor rotación de activos (0.58) —confirmado por DuPont—, lo que se traduce en el mayor ROE (24.0%) del sector. Implicancia: su ventaja competitiva actual está en eficiencia operativa y control de costos, no solo en escala.

- Laredo tiene el menor endeudamiento (34.8%) junto con un ROE sólido (9.0%). Implicancia: mayor holgura financiera para invertir o resistir choques externos sin comprometer solvencia, una posición defensiva favorable frente a volatilidad sectorial (como la observada en exportaciones de azúcar en 2024).

- Pomalca combina el ROE más bajo (1.4%) con el mayor endeudamiento relativo entre las de bajo rendimiento (50.2%) y la mayor proporción de activos no corrientes (87.5%). Implicancia: capital fuertemente inmovilizado en activos de largo plazo con retorno muy bajo, lo que sugiere baja capacidad de generar caja a corto plazo y vulnerabilidad competitiva.

---

**B.** Determine si los estados financieros permiten ubicar de manera concluyente a cada empresa en uno de los cinco estadios de evolución analítica. Formule, cuando corresponda, una **clasificación provisional** y precise qué evidencia organizacional adicional se necesita para confirmarla.

**Respuesta:** No, los estados financieros no permiten una ubicación concluyente en ningún estadio de evolución analítica, porque miden resultados económicos, no capacidades organizacionales (liderazgo, cultura de datos, tecnología, talento analítico). Como clasificación provisional, ninguna empresa puede situarse más allá de un estadio inicial-intermedio basado solo en inferencia indirecta desde sus resultados. Para confirmarlo se requeriría evidencia organizacional directa: entrevistas a gerencia, inventario de herramientas tecnológicas usadas, y evidencia de decisiones estratégicas basadas en análisis de datos.

---

**C.** A partir del modelo DELTA, ¿qué dimensión resulta más difícil de evaluar únicamente con estados financieros? Explique por qué y qué información adicional se requeriría.

**Respuesta:** La dimensión más difícil de evaluar es Leadership (o, junto a ella, Analysts), porque el compromiso directivo con la decisión basada en evidencia no deja huella en un balance o estado de resultados. Se requeriría información cualitativa: actas de directorio, entrevistas a gerencia, o evidencia de inversión sostenida en capacidades analíticas a lo largo del tiempo.

---

**D.** Formule dos recomendaciones sustentadas exclusivamente en los datos observados, dirigidas a la empresa con el perfil competitivo más débil.

**Respuesta:**

- Revisar la estructura de activos no corrientes (87.5% del activo total): evaluar si existen activos subutilizados o de baja rotación que puedan optimizarse o desinvertirse para liberar capital.

- Priorizar mejoras en margen neto (actualmente 4.1%, el más bajo del grupo) antes que en escala, dado que su rotación de activos (0.17) ya es la menor del sector — el problema no es solo volumen, sino rentabilidad por unidad vendida.

---

**E.** Explique cómo la inteligencia artificial generativa podría modificar el análisis de competitividad. Distinga entre usos plausibles y afirmaciones no verificables.

**Respuesta:** La IA generativa podría acelerar tareas plausibles como el resumen automático de estados financieros, la generación de código para calcular indicadores repetibles, o la detección de anomalías en grandes catálogos de cuentas. Sin embargo, no puede verificar por sí sola la exactitud de los datos fuente, ni emitir juicios válidos sobre causalidad financiera o madurez organizacional sin evidencia adicional — usarla para "concluir" quién es más competitivo sin sustento numérico sería una afirmación no verificable.


### Matriz de síntesis para la decisión

Complete esta matriz antes de redactar las conclusiones finales:

| Empresa | Fortaleza principal | Riesgo/debilidad principal | Evidencia numérica | Implicancia competitiva |
|---|---|---|---|---|
| Casa Grande | Liquidez razonable | Rentabilidad débil pese a mayor tamaño de activos | ROE 7.0%, ROA 3.8%, activo total 1,923,732 (el más grande del sector) | Escala no se traduce en eficiencia; capacidad de mejora en gestión de activos |
| Cartavio | Mayor rentabilidad del sector | Mayor endeudamiento y menor liquidez | ROE 24.0%, margen neto 17.2%, endeudamiento 58.6%, razón corriente 0.96 | Rentabilidad líder pero con riesgo financiero elevado |
| Agroindustrial Laredo | Menor endeudamiento con rentabilidad sólida | Rotación de activos intermedia, sin ser la más eficiente |Endeudamiento 34.8%, ROE 9.0%, rotación 0.47 | Perfil más equilibrado riesgo-rentabilidad del sector |
| Pomalca | Mayor liquidez del grupo | Rentabilidad casi nula y mayor concentración en activos no corrientes | ROE 1.4%, margen neto 4.1%, 87.5% de activos no corrientes, razón corriente 1.55 | Perfil competitivo más débil; capital inmovilizado con bajo retorno |

Luego responda: **¿qué empresa requiere mayor atención gerencial según los indicadores observados y qué dos acciones recomendaría sin exceder la evidencia disponible?**

**Respuesta:**

Pomalca requiere la mayor atención gerencial: presenta el ROE más bajo del sector (1.4%), el margen neto más bajo (4.1%) y la mayor proporción de activos inmovilizados en el largo plazo (87.5%), una combinación que indica capital de baja productividad.

Dos acciones sin exceder la evidencia disponible:

- Evaluar la eficiencia de sus activos no corrientes, dado que un margen neto y una rotación tan bajos (0.17) sugieren capacidad instalada subutilizada respecto a las ventas generadas.

- Revisar su estructura de costos operativos, ya que su margen neto (4.1%) es muy inferior al del resto del sector pese a contar con la mayor razón corriente (1.55), lo que descarta un problema de liquidez inmediata como causa principal de su baja rentabilidad.

---

## Reto de aplicación y retroalimentación

En esta sección se aplicarán los procedimientos desarrollados durante la sesión a nuevas situaciones de análisis. Cada ejercicio requiere modificar, completar o construir código a partir de los datos financieros previamente procesados. Posteriormente, los resultados obtenidos deberán interpretarse brevemente desde una perspectiva empresarial. El propósito es comprobar la comprensión de las técnicas utilizadas y fortalecer la capacidad de adaptar el análisis ante nuevas preguntas de negocio.

**Indicaciones generales.** Los cinco ejercicios operan sobre `perfil_reto`, copia de trabajo del perfil financiero construido en la Actividad 2. Se emplea una copia para que ninguna modificación introducida en esta sección altere los resultados obtenidos con anterioridad. Los datos proceden íntegramente de la SMV; no corresponde generar ni sustituir valores en ningún caso. Cada respuesta escrita no debe exceder cuatro líneas.

**Duración en sesión:** 25 minutos. Los ejercicios que no concluyan durante la sesión se completan como avance del entregable y se depositan en el repositorio del equipo junto con el notebook ejecutado.


In [12]:
# Copia de trabajo para la sección de retos.
perfil_reto = perfil.copy()
print("Copia de trabajo creada:", perfil_reto.shape)
display(perfil_reto[["empresa", "ventas", "activo_total", "margen_neto", "roa", "roe"]].round(3))


Copia de trabajo creada: (4, 14)


,empresa,ventas,activo_total,margen_neto,roa,roe
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,940324.0,0.126,0.059,0.090
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,926161.0,0.172,0.099,0.240
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,1923732.0,0.115,0.038,0.070
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,1352469.0,0.041,0.007,0.014


### Ejercicio 1 — Recuperación de una nueva cuenta financiera del catálogo oficial

Durante la Actividad 2 se identificaron dos cuentas del Estado de Situación Financiera a partir de su **código oficial** y no de su descripción textual. Ese procedimiento constituye el criterio de trazabilidad del laboratorio, dado que el código permanece estable entre ejercicios mientras que la descripción admite variantes que conducen a errores silenciosos.

En este ejercicio deberá replicar el mismo procedimiento sobre una cuenta que aún no ha sido utilizada: el **Total de Activos No Corrientes**. Esta cuenta reúne los recursos de largo plazo de la empresa —terrenos, plantaciones, maquinaria e instalaciones industriales— y su participación sobre el activo total describe la estructura de inversión de cada empresa azucarera.

**Se solicita:**

1. Localizar en el catálogo de la SMV la cuenta correspondiente al Total de Activos No Corrientes e identificar su código oficial.
2. Recuperar el valor de esa cuenta para las cuatro empresas analizadas.
3. Incorporar la variable a `perfil_reto`.
4. Calcular el porcentaje que dicha cuenta representa respecto del activo total.

**Tiempo estimado:** 10 a 12 minutos.


In [13]:
# Paso A: localizar el código en el catálogo
muestra_e1 = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra_e1, ["descripcion_cuenta", "descripcion"])
catalogo_no_corriente = muestra_e1[
    muestra_e1[columna_desc].astype(str).str.contains("o Corrientes", case=False, na=False)
][["cuenta", columna_desc, "monto1"]]
display(catalogo_no_corriente.reset_index(drop=True))

,cuenta,descripcion_cuenta,monto1
0,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
1,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
2,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
3,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
4,1D02ST,Total Activos No Corrientes,802358.0
5,1D04ST,Total Pasivos No Corrientes,202039.0


In [14]:
# Paso B: reemplazar con el código que identificaste en el catálogo
TOTAL_ACTIVO_NO_CORRIENTE = "1D02ST"

def obtener_cuenta(df, codigo, columna_valor="monto1"):
    col_cuenta = resolver_columna(df, ["cuenta"])
    resultado = df[df[col_cuenta] == codigo][["nombre_empresa", columna_valor]].copy()
    resultado = resultado.rename(columns={"nombre_empresa": "empresa", columna_valor: "activo_no_corriente"})
    resultado["activo_no_corriente"] = resultado["activo_no_corriente"].map(a_numero)
    return resultado

activo_no_corriente = obtener_cuenta(balance, TOTAL_ACTIVO_NO_CORRIENTE)
perfil_reto = perfil_reto.merge(activo_no_corriente, on="empresa", how="left")
perfil_reto["pct_activo_no_corriente"] = perfil_reto["activo_no_corriente"] / perfil_reto["activo_total"]

display(perfil_reto[["empresa", "activo_no_corriente", "activo_total", "pct_activo_no_corriente"]].round(3))

,empresa,activo_no_corriente,activo_total,pct_activo_no_corriente
0,AGROINDUSTRIAL LAREDO S.A.A.,802358.0,940324.0,0.853
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,631228.0,926161.0,0.682
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1494428.0,1923732.0,0.777
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1183634.0,1352469.0,0.875


**Pregunta 4.** ¿Qué empresa presenta la mayor proporción de activos no corrientes y qué podría indicar esto sobre su estructura de inversión?

**Respuesta (máximo cuatro líneas):** Agroindustrial Pomalca S.A.A. tiene la mayor participación de activos no corrientes, con 87.5% del total, seguida por Agroindustrial Laredo con 85.3%. Esto muestra que gran parte de sus recursos está invertida en elementos de largo plazo, como terrenos, plantaciones y maquinaria, dejando una proporción menor en activos de corto plazo. Cartavio presenta una estructura relativamente más equilibrada, con 68.2% de activos no corrientes.



### Ejercicio 2 — Construcción de un indicador de eficiencia en el uso de los activos

Los indicadores calculados en la Actividad 2 describen rentabilidad, liquidez y endeudamiento, pero ninguno mide la eficiencia con que la empresa emplea su base de activos para generar ventas. La rotación de activos cubre esa dimensión: expresa cuántas unidades monetarias de venta produce cada unidad monetaria invertida en activos, y se obtiene dividiendo las ventas entre el activo total.

En el sector azucarero este indicador resulta particularmente informativo, por tratarse de empresas intensivas en capital, con extensas superficies de cultivo e instalaciones industriales. Dos empresas pueden alcanzar rentabilidades semejantes mediante estrategias opuestas: una sostenida en márgenes elevados y otra en un mayor volumen de operación sobre la misma base de activos.

**Se solicita:**

1. Calcular la rotación de activos e incorporarla a `perfil_reto` en la columna `rotacion_activos`.
2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
3. Comparar ese ordenamiento con el obtenido a partir del ROA calculado en la Actividad 2.

**Tiempo estimado:** 10 a 12 minutos.


In [15]:
perfil_reto["rotacion_activos"] = perfil_reto["ventas"] / perfil_reto["activo_total"]

ranking_rotacion = perfil_reto[["empresa", "rotacion_activos"]].sort_values(
    "rotacion_activos", ascending=False
)
print("Ranking por rotación de activos:")
display(ranking_rotacion)

ranking_roa_e2 = perfil_reto[["empresa", "roa"]].sort_values("roa", ascending=False)
print("Ranking por ROA (referencia de la Actividad 2):")
display(ranking_roa_e2)

Ranking por rotación de activos:


,empresa,rotacion_activos
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.577733
0,AGROINDUSTRIAL LAREDO S.A.A.,0.466790
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.331481
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.173050


Ranking por ROA (referencia de la Actividad 2):


,empresa,roa
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.099169
0,AGROINDUSTRIAL LAREDO S.A.A.,0.058972
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.038202
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.007125


**Pregunta 5.** ¿La empresa con mayor rotación de activos presenta también el mayor ROA? Explique brevemente a partir de los resultados obtenidos.

**Respuesta (máximo cuatro líneas):** Sí. Cartavio ocupa el primer lugar tanto en rotación de activos (0.58 veces) como en ROA (9.9%). Además, el orden de las empresas se mantiene igual en ambos indicadores: Cartavio, Laredo, Casa Grande y Pomalca. Esto me indica que, en este grupo, la capacidad de aprovechar los activos para generar ventas tiene una relación importante con el nivel de rentabilidad obtenido sobre esos activos.



### Ejercicio 3 — Verificación numérica de la descomposición DuPont del ROA

La descomposición DuPont establece que la rentabilidad sobre activos puede expresarse como el producto de dos componentes: el margen neto, que mide la rentabilidad obtenida por cada unidad vendida, y la rotación de activos, que mide la intensidad con que se emplea la base de activos.

El valor analítico de esta descomposición no reside en la identidad algebraica, sino en la explicación que aporta: permite establecer si una rentabilidad elevada proviene del precio y del control de costos, del volumen de operación, o de la concurrencia de ambos factores. Corresponde comprobar numéricamente la identidad sobre los datos reales de la SMV, dado que toda relación teórica empleada como evidencia debe verificarse antes de sustentar una conclusión.

**Se solicita:**

1. Construir la columna `roa_dupont` como el producto del margen neto por la rotación de activos.
2. Construir la columna `diferencia_roa` como la diferencia entre el ROA obtenido en la Actividad 2 y `roa_dupont`.
3. Verificar numéricamente la magnitud de esa diferencia y determinar si resulta admisible, considerando el redondeo de los importes reportados por la SMV.

**Tiempo estimado:** 10 a 12 minutos.


In [16]:
perfil_reto["roa_dupont"] = perfil_reto["margen_neto"] * perfil_reto["rotacion_activos"]
perfil_reto["diferencia_roa"] = perfil_reto["roa"] - perfil_reto["roa_dupont"]

display(perfil_reto[["empresa", "roa", "roa_dupont", "diferencia_roa"]].round(6))
print("Diferencia máxima observada:", perfil_reto["diferencia_roa"].abs().max())

,empresa,roa,roa_dupont,diferencia_roa
0,AGROINDUSTRIAL LAREDO S.A.A.,0.058972,0.058972,-0.0
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.099169,0.099169,0.0
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.038202,0.038202,0.0
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.007125,0.007125,0.0


Diferencia máxima observada: 6.938893903907228e-18


**Pregunta 6.** ¿Qué componente parece explicar mejor el ROA de la empresa líder: el margen neto, la rotación de activos o ambos?

**Respuesta (máximo cuatro líneas):** Al revisar el margen neto implícito, Cartavio alcanza aproximadamente 17.2%, frente a 12.6% de Laredo, 11.5% de Casa Grande y 4.1% de Pomalca. Los dos factores benefician a Cartavio, pero considero que el margen neto es el que marca una diferencia mayor respecto a Laredo, ya que sus rotaciones son relativamente cercanas (0.58 frente a 0.47). Por ello, el margen parece ser el principal elemento que explica su liderazgo en ROA, aunque la eficiencia de sus activos también contribuye.



### Ejercicio 4 — Consulta empresarial con condiciones múltiples en DuckDB

En la Actividad 2 se empleó DuckDB para reproducir un ordenamiento ya obtenido con Polars, con la finalidad de comprobar la reproducibilidad del resultado entre dos motores distintos. Una consulta de negocio, sin embargo, no se limita a ordenar: delimita un subconjunto de la población mediante criterios definidos previamente por la gerencia.

En este ejercicio deberá construir una consulta que identifique aquellas empresas que satisfacen **simultáneamente** dos condiciones financieras —capacidad de atender obligaciones de corto plazo y generación de rentabilidad positiva para el accionista— y que presente el resultado ordenado según su exposición al endeudamiento. Este es el procedimiento habitual mediante el cual un área de riesgos delimita una cartera de empresas elegibles.

**Se solicita:**

1. Construir sobre `perfil_reto` una consulta SQL que incluya `SELECT`, `WHERE`, condiciones múltiples enlazadas con `AND` y `ORDER BY`.
2. Establecer como criterios una razón corriente mayor o igual a 1 y un ROE positivo.
3. Ordenar el resultado por endeudamiento, de menor a mayor.
4. Determinar cuántas empresas satisfacen ambos criterios.

**Tiempo estimado:** 10 a 12 minutos.


In [17]:
consulta_reto4 = duckdb.sql("""
    SELECT empresa,
           ROUND(razon_corriente, 2)      AS razon_corriente,
           ROUND(roe * 100, 1)            AS roe_pct,
           ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct
    FROM perfil_reto
    WHERE razon_corriente >= 1
      AND roe > 0
    ORDER BY endeudamiento_pct ASC
""").df()

display(consulta_reto4)
print("Número de empresas que cumplen ambos criterios:", len(consulta_reto4))


,empresa,razon_corriente,roe_pct,endeudamiento_pct
0,AGROINDUSTRIAL LAREDO S.A.A.,1.10,9.0,34.8
1,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1.18,7.0,45.4
2,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1.55,1.4,50.2


Número de empresas que cumplen ambos criterios: 3


**Pregunta 7.** Según los criterios establecidos, ¿qué empresa presenta la condición financiera más favorable dentro del grupo analizado?


**Respuesta (máximo cuatro líneas):** Agroindustrial Laredo presenta la posición más favorable porque cumple los dos criterios considerados: tiene el menor endeudamiento (34.8%) y mantiene un ROE competitivo de 9.0%. Casa Grande y Pomalca también cumplen, pero presentan mayores niveles de deuda y, en el caso de Pomalca, una rentabilidad bastante reducida (1.4%). Cartavio no aparece entre las empresas que cumplen las condiciones, probablemente porque su razón corriente es menor a 1.



### Ejercicio 5 — Representación conjunta de la rentabilidad y del nivel de endeudamiento

Los gráficos de barras construidos en la Actividad 2 comparan una dimensión por vez. La decisión empresarial, sin embargo, rara vez depende de un solo indicador: una rentabilidad elevada sostenida sobre un endeudamiento igualmente elevado no equivale a una rentabilidad semejante obtenida con una estructura financiera conservadora, aun cuando ambas presenten el mismo ROE.

El gráfico de dispersión permite examinar dos dimensiones de manera simultánea y, mediante el tamaño del marcador, incorporar una tercera. Al situar el endeudamiento en el eje horizontal y el ROE en el eje vertical, cada cuadrante adquiere una lectura gerencial precisa: la combinación de rentabilidad elevada con endeudamiento reducido corresponde al perfil más favorable, mientras que la combinación inversa exige un examen detenido.

**Se solicita:**

1. Construir con Plotly un gráfico de dispersión que sitúe el endeudamiento en el eje horizontal y el ROE en el eje vertical.
2. Identificar cada punto con el nombre de la empresa correspondiente.
3. Representar el activo total mediante el tamaño del marcador.
4. Rotular los ejes y titular el gráfico de modo que resulte interpretable sin recurrir al código.

**Tiempo estimado:** 10 a 12 minutos.


In [18]:
fig_disp = go.Figure(go.Scatter(
    x=perfil_reto["endeudamiento"] * 100,
    y=perfil_reto["roe"] * 100,
    mode="markers+text",
    text=perfil_reto["empresa"],
    textposition="top center",
    marker=dict(
        size=perfil_reto["activo_total"],
        sizemode="area",
        sizeref=2. * perfil_reto["activo_total"].max() / (60.**2),
        sizemin=8,
        color="#00A9CE",
        line=dict(width=1, color="white"),
    ),
))
fig_disp.update_layout(
    title="ROE vs. Endeudamiento (tamaño del marcador = activo total) — sector azucarero",
    xaxis_title="Endeudamiento (%)",
    yaxis_title="ROE (%)",
    height=520,
    width=900,
    template="plotly_white",
)
fig_disp.show()

**Pregunta 8.** ¿Qué empresa muestra la combinación más favorable entre rentabilidad y nivel de endeudamiento?

**Respuesta (máximo cuatro líneas):** Agroindustrial Laredo presenta, en mi opinión, el mejor equilibrio, porque combina un ROE de 9% con el menor endeudamiento del grupo, cercano al 35%. Cartavio consigue el ROE más alto (24%), pero también tiene el mayor endeudamiento (59%), por lo que su rentabilidad está acompañada de una mayor exposición financiera. Casa Grande y Pomalca muestran una combinación menos atractiva al tener una deuda relativamente alta y una rentabilidad menor.



### Retroalimentación final — Vinculación con el modelo DELTA Plus

Los cinco ejercicios anteriores se resolvieron mediante Python, DuckDB y Plotly sobre datos financieros oficiales. Corresponde ahora situar ese trabajo dentro del marco conceptual de la sesión. Responda de manera breve.

**Pregunta 9.** A partir del análisis realizado mediante Python, identifique qué dimensión de DELTA Plus se evidencia directamente en esta actividad: *Technology*, *Analytical Techniques* o ambas. Justifique su respuesta en un máximo de tres líneas.

**Respuesta:** Se evidencian ambas dimensiones: Technology, porque el análisis usa infraestructura de procesamiento (Python, Polars, DuckDB) y visualización (Plotly) sobre datos reales; y Analytical Techniques, porque se aplicaron técnicas descriptivas (rankings, indicadores) y diagnósticas (descomposición DuPont) para explicar el desempeño.

---

**Pregunta 10.** *Technology.* ¿Qué tecnologías resultarían necesarias para que una de las empresas analizadas pudiera ejecutar de manera periódica y reproducible el análisis desarrollado en este laboratorio?

**Respuesta (máximo cuatro líneas):** Necesitaría automatizar la descarga periódica desde la API de la SMV (en lugar de ejecución manual), un entorno reproducible (notebooks versionados o pipeline en la nube), almacenamiento estructurado tipo data warehouse para conservar históricos, y un tablero de visualización (dashboard) que actualice los indicadores sin reescribir código cada vez.

---

**Pregunta 11.** *Analytical Techniques.* ¿En qué nivel de técnicas analíticas se sitúa el análisis realizado en esta sesión y qué técnica adicional podría incorporar la organización para mejorar sus decisiones?

**Respuesta (máximo cuatro líneas):** El análisis se sitúa principalmente en el nivel diagnóstico (explica por qué ocurre el ROA mediante DuPont), con elementos descriptivos (rankings, comparaciones). Para avanzar, la organización podría incorporar técnicas predictivas, por ejemplo proyectar el ROE o el riesgo de liquidez ante escenarios de tipo de cambio o precio del azúcar usando los datos del BCRP.

---


## Conclusiones

1. Al comparar los resultados obtenidos con Polars, DuckDB y la fórmula de DuPont, pude comprobar que los cálculos son consistentes y se pueden reproducir. La diferencia entre el ROA calculado directamente y el obtenido mediante DuPont fue prácticamente cero (~7×10⁻¹⁸), lo que respalda el uso de los datos oficiales de la SMV.

2. Cartavio destaca por sus niveles de rentabilidad, con un ROE de 24% y un ROA de 9.9%, y una parte importante de esta ventaja se relaciona con su margen neto. Sin embargo, también presenta el mayor endeudamiento, cercano al 59%. En cambio, Laredo me parece la empresa con el equilibrio financiero más favorable, al combinar un ROE aproximado de 9% con el menor nivel de deuda, alrededor del 35%.

3. El análisis también muestra que el sector todavía tiene oportunidades para avanzar en madurez analítica, especialmente en el uso de tecnología y técnicas analíticas. Considero que automatizar estos indicadores y complementar la información financiera con variables del BCRP, como el tipo de cambio y las exportaciones, permitiría desarrollar análisis predictivos y tomar decisiones con mayor respaldo de datos.

## Referencias oficiales

- Superintendencia del Mercado de Valores. Portal de Datos Abiertos y servicio de información financiera.
- Banco Central de Reserva del Perú. Series estadísticas mensuales.
